exporter/prometheusremotewriteexporter/exporter.go

```diff
@@ -107,7 +109,7 @@ func (prwe *prwExporter) PushMetrics(ctx context.Context, md pdata.Metrics) erro
 	case <-prwe.closeChan:
 		return errors.New("shutdown has been called")
 	default:
-		tsMap, err := prometheusremotewrite.FromMetrics(md, prometheusremotewrite.Settings{Namespace: prwe.namespace, ExternalLabels: prwe.externalLabels})
+		tsMap, err := prometheusremotewrite.FromMetrics(md, prometheusremotewrite.Settings{Namespace: prwe.namespace, ExternalLabels: prwe.externalLabels, SanitizeLabel: prwe.sanitizeLabel})
```

Comment:
Can this be done without changing the exported interface of the PRW translator package?

Reply:
I don't see much options, one is to introduce a feature flag in the translator package, but I was thinking to reuse the same feature flag as basically it's about the same thing as #7112 , thus an additional fields in `prometheusremotewrite.Settings` will be introduced to follow previous behavior by default.

Reply:
The thing is that feature gates are intended to be temporary and removed eventually.  Changes to the public API of a package are not reversible (assuming a world where we're v1.x, which we need to start planning for) and thus should be avoided in conjunction with feature gates.  It is hypothesized that the sanitization of labels starting with `_` (but not `__`) is unnecessary, so perhaps we should focus on finding a way to gain confidence in that hypothesis and eliminating the need for this feature gate.

Reply:
I understand the concern of changing the public API in translator, but it's still the same behavior by default, with regards to gain confidence in the hypothesis, I am not sure how we can move forward on that. could you please elaborate?

Reply:
hi @Aneurysm9  any thoughts on how we can move forward on this?

Reply:
@Aneurysm9 I see there was similar [discussion](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/7112#issuecomment-1007887902 ) in #7112. I had e2e testing in prometheus with following config

```yaml
  prometheus.yml: |
    remote_write:
      - url: "http://victoriametrics:8428/api/v1/write"
    remote_read:
      - url: "http://victoriametrics:8428/api/v1/read"
        read_recent: true

    scrape_configs:
      - job_name: "prometheus"
        static_configs:
          - targets: ["localhost:9090"]
            labels:
              _namespace_: test
```

from the query result, `_namespace_` label is not actually sanitized

![image](https://user-images.githubusercontent.com/14903757/159215355-f9b15e71-2421-4eb7-ab63-30ec72018d36.png)

Reply:
I would also think code is copied/pasted into this repo as @vjsamuel mentioned, I see similar code for sanitize in [opencensus](https://github.com/census-instrumentation/opencensus-go/blob/master/internal/sanitize.go).

Reply:
Then we should probably eliminate it altogether rather than further propagating it.  Changing the public API of this module does not seem like the correct thing to do.

I'm not sure I'm comfortable making this decision by myself.  @dashpole @bogdandrutu @jpkrohling @codeboten @gouthamve

Reply:
yeah that sounds like this will be a breaking change which is going to remove the `key` prefix.

If not to change the public API, would it work that introduce a new method similar to `FromMetrics` which does not sanitize label(no `key` prefix)?

Reply:
>  Changing the public API of this module does not seem like the correct thing to do.

I share the same opinion.

Reply:
I also agree we should not change the public API.  We should either make the change outright, or use a feature-gate to slowly make the change.

Reply:
Added a feature gate for this.